assets/javascripts/kitten/components/uploaders/uploader.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import classNames from 'classnames'
+
+export class Uploader extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      errors: {
+        extensionsError: false,
+        sizeError: false,
+        hasError: false,
+      },
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  checkExtensions(file) {
+    if (!this.props.acceptedFiles) return
+
+    const extensions = this.props.acceptedFiles.split(',')
+    const regex = new RegExp(`(${ extensions.join('|') })`)
+
+    this.setState({
+      errors: {
+        ...this.state.errors,
+        extensionsError: file.type.search(regex) < 1,
+      }
+    })
+  }
+
+  checkSize(file) {
+    this.setState({
+      errors: {
+        ...this.state.errors,
```

Comment:
Je crois que ce n'est pas vraiment recommandé de se servir de `this.state` pour refaire un `setState`. Peut-être qu'il faudrait mieux avoir juste un état `this.state.sizeError` ?